stargz/registargzify/registargzify.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2019 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// The stargzify command converts a tarball into a seekable stargz
+// tarball. The output is still a valid tarball, but has new gzip
+// streams throughout the file and and an Table of Contents (TOC)
+// index at the end pointing into those streams.
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/google/crfs/stargz"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/empty"
+	"github.com/google/go-containerregistry/pkg/v1/mutate"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/google/go-containerregistry/pkg/v1/tarball"
+)
+
+var (
+	in  = flag.String("in", "", "image to be stargzified.")
+	out = flag.String("out", "", "destination for that image.")
+)
+
+func main() {
+	flag.Parse()
+	src, dst := *in, *out
+	srcRef, err := name.ParseReference(src, name.WeakValidation)
```

Comment:
Will this still support local files on disk?

Reply:
I could support both files and images but I'm struggling to find a flag/argument scheme that makes using the tool not awkward. That's why I originally had it as a separate tool.